### PR TITLE
Add some clarifying documentation for pull through cache configuration

### LINKED ISSUE #922: Pull Through Cache For Private Registry
I followed instruction on how to setup [Pull Through Cache] (https://github.com/docker/distribution/blob/master/docs/mirror.md) but it turned out that pull through cache is not yet supported for private registry yet. I think the documentation needs to be updated as this restriction is not mentioned anywhere. I also think that it would be nice to be able to have a pull through cache that can proxy a private registry and not just Docker hub.

### PR DESCRIPTION
**Pull request by RichardScothern**
Closes #922 #927

Signed-off-by: Richard Scothern <richard.scothern@gmail.com>

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -1940,6 +1944,59 @@ The TCP address to connect to, including a port number.
...
 </table>
 
+## Proxy
+
+    proxy:
+      remoteurl: https://registry-1.docker.io
+      username: [username]
+      password: [password]
+
+Proxy enables a registry to be configured as a pull through cache to the official Docker Index.  See [mirror.md](mirror.md) for more information
```
Comment (dmp42) [inline docs/configuration.md]: s/Index/Hub/

File: docs/mirror.md [hunk 2]
```diff
@@ -8,6 +8,8 @@ draft = "true"
 
 A v2 Registry can be configured as a pull through cache.  In this mode a Registry responds to all normal docker pull requests but stores all content locally.
 
+NOTE: Currently this feature can only be used to proxy against the official Docker Index
```
Comment (dmp42) [inline docs/mirror.md]: Index -> Hub

File: docs/configuration.md [hunk 3]
```diff
@@ -1940,6 +1944,59 @@ The TCP address to connect to, including a port number.
...
+  </tr>
+  <tr>
+    <td>
+      <code>remoteurl</code>
+    </td>
+    <td>
+      yes
+    </td>
+    <td>
+     The URL of the official Docker Index
```
Comment (dmp42) [inline docs/configuration.md]: Index -> Hub

File: docs/configuration.md [hunk 4]
```diff
@@ -1940,6 +1944,59 @@ The TCP address to connect to, including a port number.
...
+  </tr>
+  <tr>
+    <td>
+      <code>username</code>
+    </td>
+    <td>
+      no
+    </td>
+    <td>
+     The username of the Docker Index account
```
Comment (dmp42) [inline docs/configuration.md]: Hub

File: docs/configuration.md [hunk 5]
```diff
@@ -1940,6 +1944,59 @@ The TCP address to connect to, including a port number.
...
+  </tr>
+  <tr>
+    <td>
+      <code>password</code>
+    </td>
+    <td>
+      no
+    </td>
+    <td>
+     The password for the official Docker Index account
```
Comment (dmp42) [inline docs/configuration.md]: Hub

File: docs/configuration.md [hunk 6]
```diff
@@ -1940,6 +1944,59 @@ The TCP address to connect to, including a port number.
...
+    </td>
+  </tr>
+</table>
+
+To enable pulling private repositories (e.g. `batman/robin`) a username and password for user `batman` must be specified.
```
Comment (dmp42) [inline docs/configuration.md]: A word about what that means in term of privacy of the content is probably welcome.

"By allowing your mirror to access your private resources on the Hub, you are delegating the responsibility to secure access to these resources to the mirror itself. Consider yourself warned, cowboy."

Or something like that in a more appropriate language :)

File: docs/configuration.md [hunk 7]
```diff
@@ -1940,6 +1944,59 @@ The TCP address to connect to, including a port number.
   </tr>
 </table>
 
+## Proxy
+
+    proxy:
+      remoteurl: https://registry-1.docker.io
```
Comment (dmp42) [inline docs/configuration.md]: We should have a better domain name for it, but meh.

**Comment by dmp42** [comment]
Nits. LGTM otherwise.

**Comment by stevvooe** [comment]
@RichardScothern How does this close #922?

**Comment by dmp42** [comment]
Reopened, and closed #927

**Pull request merged**